app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -107,29 +112,82 @@ fun Nav() {
             composable(route = Screen.Reminder.route + "/{dateString}") {
                 val reminderViewModel = hiltViewModel<ReminderViewModel>()
                 val state by reminderViewModel.state.collectAsState()
+                val editedRemindDescription = navController
+                    .currentBackStackEntry
+                    ?.savedStateHandle
+                    ?.getStateFlow<String>("reminderDescription", "Reminder Description")
+                    ?.collectAsStateWithLifecycle()
+
+                LaunchedEffect(editedRemindDescription) {
+                    reminderViewModel.setReminderDescription(
+                        editedRemindDescription?.value ?: "Reminder Description"
+                    )
+                }
```

Comment:
Ah, you might need to refer to `editedRemindDescription.value` here for the key of LaunchedEffect as well, otherwise Compose will listen to changes of the whole state field and not its value.
Alternatively, you can use `by` to avoid having to refer to `.value` all the time: 
```
val editedReminderDescription by navController.currentBackStackEntry...
```